sync-android/src/main/java/com/cloudant/sync/sqlite/android/AndroidSQLCipherSQLite.java

```diff
@@ -0,0 +1,233 @@
+
+    /**
+     * Constructor for opening SQLCipher-based SQLite database.
+     * @param path full file path of the db file
+     * @param provider Provider object that contains the key to encrypt the SQLCipher database
+     * @return
+     */
+    public static AndroidSQLCipherSQLite openAndroidSQLite(String path, KeyProvider provider) {
+
+        //Call SQLCipher-based method for opening (or creating) database
+        SQLiteDatabase db = SQLiteDatabase.openOrCreateDatabase(path,
+                KeyUtils.sqlCipherKeyForKeyProvider(provider), null);
+
+        return new AndroidSQLCipherSQLite(db);
+    }
+
+    public AndroidSQLCipherSQLite(final SQLiteDatabase database) {
+        this.database = database;
+        this.threadWhichOpened = Thread.currentThread();
+        this.appearsOpen = new ThreadLocal<Boolean>(){
+            @Override
+            protected Boolean initialValue() {
+                return database.isOpen();
+            }
+        };
+    }
+
+    @Override
+    public void compactDatabase() {
+        database.execSQL("VACUUM");
+    }
+
+    @Override
+    public void open() {
+        // database should be already opened
+    }
+
+    @Override
+    public void close() {
+        // Since the JavaSE version of this needs to have each thread that operated on the db
+        // close its connection to the db. Since android does not have ThreadLocal connections
+        // the connection only needs to be closed once.
+        // To maintain compatibility with JavaSE unless the current thread opened the connection,
+        // the db only appears closed to the thread that called close
+        if(threadWhichOpened == Thread.currentThread())
+            this.database.close();
+        appearsOpen.set(Boolean.FALSE);
+    }
+
+    // This implementation of isOpen will only return true if the database is open AND the current
+    // thread has not called the close() method on this object previously, this makes it compatible
+    // with the JavaSE SQLiteWrapper, where each thread closes its own connection.
+    @Override
+    public boolean isOpen() {
+        return this.database.isOpen() && this.appearsOpen.get();
+    }
+
+    @Override
+    public void beginTransaction() {
+        this.database.beginTransaction();
+    }
+
+    @Override
+    public void endTransaction() {
+        this.database.endTransaction();
+    }
+
+    @Override
+    public void setTransactionSuccessful() {
+        this.database.setTransactionSuccessful();
+    }
+
+    @Override
+    public void execSQL(String sql)  {
+        Preconditions.checkArgument(!Strings.isNullOrEmpty(sql.trim()),
+                "Input SQL can not be empty String.");
+        this.database.execSQL(sql);
+    }
+
+    public int status(int operation, boolean reset) {
+        return 0;
+    }
+
+    //TODO: Change password option that will be implemented with key management work items
+    public void changePassword(String password) throws SQLiteException {
```

Comment:
Just for the record, this method is not part of the iOS API. It is a task recorded in FB but the method is not there yet, anyway it will be probably part of `CDTEncryptionKeychainManager`.